### ISSUE TITLE
Issue with distributed processing

### ISSUE THREAD

**Issue by MohamedOmar2020** [Dana-Farber-AIOS/pathml#164]
Hello, 
Thank you for fixing the distributed issue with the mesmer model. I am running the pipeline with 'distributed = True' flag but I am getting many warnings and errors. Additionally, the pipeline was supposed to return 145 tiles but it is returning only 3 !.
This is a part of the log message:

> def watershed(image, markers=None, connectivity=1, offset=None, mask=None,
> /Users/mohamedomar/.local/lib/python3.8/site-packages/skimage/morphology/_deprecated.py:5: skimage_deprecation: Function ``watershed`` is deprecated and will be removed in version 0.19. Use ``skimage.segmentation.watershed`` instead.
>   def watershed(image, markers=None, connectivity=1, offset=None, mask=None,
> /Users/mohamedomar/opt/anaconda3/envs/pathml2/lib/python3.8/site-packages/anndata/_core/anndata.py:120: ImplicitModificationWarning: Transforming to str index.
>   warnings.warn("Transforming to str index.", ImplicitModificationWarning)
> /Users/mohamedomar/.local/lib/python3.8/site-packages/skimage/morphology/_deprecated.py:5: skimage_deprecation: Function ``watershed`` is deprecated and will be removed in version 0.19. Use ``skimage.segmentation.watershed`` instead.
>   def watershed(image, markers=None, connectivity=1, offset=None, mask=None,
> /Users/mohamedomar/opt/anaconda3/envs/pathml2/lib/python3.8/site-packages/anndata/_io/h5ad.py:64: FutureWarning: The `force_dense` argument is deprecated. Use `as_dense` instead.
>   warn(
> /Users/mohamedomar/opt/anaconda3/envs/pathml2/lib/python3.8/site-packages/anndata/_core/anndata.py:120: ImplicitModificationWarning: Transforming to str index.
>   warnings.warn("Transforming to str index.", ImplicitModificationWarning)
>  storing 'coords' as categorical
>  storing 'slice' as categorical
>  storing 'tile' as categorical
**_**> 2021-08-10 00:00:05.176962: W tensorflow/core/framework/op_kernel.cc:1763] OP_REQUIRES failed at save_restore_v2_ops.cc:205 : Out of range: Read less bytes than requested
> distributed.worker - WARNING - Compute Failed
> Function:  apply
> args:      (Tile(coords=(1598, 6616),
> 	name=None,
> 	image shape: (1598, 1654, 2),
> 	slide_type=SlideType(stain=Fluor, platform=Vectra, tma=None, rgb=None, volumetric=None, time_series=None),
> 	labels=None,
> 	masks=None,
> 	counts=None))
> kwargs:    {}
> Exception: OutOfRangeError()**_**
 
That last error (bold text) is repeated many times. 

Thanks in advance

**Comment by ryanccarelli** [comment]
Is this on SCU using slurm? https://github.com/tensorflow/tensorflow/issues/21544#issuecomment-418848591 suggests that you didn't request enough memory

**Comment by ryanccarelli** [comment]
Actually I think this issue means disk space not RAM

**Comment by MohamedOmar2020** [comment]
That seems unlikely since I have 135 GB free and I have the same problem on Slurm. 
Can you reproduce it? These are my package versions:
tensorflow: 2.4.1
dask: 2021.7.1    
distributed: 2021.7.1

**Comment by MohamedOmar2020** [comment]
Closing this since I found a fix on my end. 
For those encountering a similar problem in the future, the 'out of range' error is because the tile shape is not dividable by the number of workers.

For example, the following code will result in an error:

```
cluster = LocalCluster(n_workers=16, threads_per_worker=1, processes=True)
client = Client(cluster)
slidedata.run(pipe, distributed = True, client = client, tile_size= (1200, 1800), tile_pad=False, overwrite_existing_tiles=True)
```
However, it will work smoothly with 'n_workers=10' since the tile will be divided into 10 parts each with the shape of (120, 180) /worker.
